airbyte-webapp/src/views/Connector/ConnectorCard/ConnectorCard.tsx

```diff
@@ -93,26 +85,47 @@ export const ConnectorCard: React.FC<ConnectorCardCreateProps | ConnectorCardEdi
       setSaved(true);
     } catch (e) {
       setErrorStatusRequest(e);
+      setIsFormSubmitting(false);
     }
   };
 
   const job = jobInfo || LogsRequestError.extractJobInfo(errorStatusRequest);
 
+  const { selectedConnectorDefinitionSpecification, onServiceSelect, availableServices, isEditMode } = props;
+  const selectedConnectorDefinitionSpecificationId =
+    selectedConnectorDefinitionSpecification && ConnectorSpecification.id(selectedConnectorDefinitionSpecification);
+
   return (
-    <Card title={title} fullWidth={full}>
-      <ServiceForm
-        {...props}
-        errorMessage={props.errorMessage || (error && generateMessageFromError(error))}
-        isTestConnectionInProgress={isTestConnectionInProgress}
-        onStopTesting={onStopTesting}
-        testConnector={testConnector}
-        onSubmit={onHandleSubmit}
-        successMessage={
-          props.successMessage || (saved && props.isEditMode && <FormattedMessage id="form.changesSaved" />)
-        }
-      />
-      {/* Show the job log only if advanced mode is turned on or the actual job failed (not the check inside the job) */}
-      {job && (advancedMode || !job.succeeded) && <JobItem job={job} />}
-    </Card>
+    <>
+      <Card title={title} fullWidth={full} className={styles.serviceTypeSelectCard}>
+        <div className={styles.dropdownContainer}>
+          <ConnectorServiceTypeControl
+            formType={props.formType}
+            onChangeServiceType={onServiceSelect}
+            availableServices={availableServices}
+            isEditMode={isEditMode}
+            selectedServiceId={selectedConnectorDefinitionSpecificationId}
+            disabled={isFormSubmitting}
+          />
+          {additionalDropdownComponent}
+        </div>
+      </Card>
+      {intermediateComponent}
```

Comment:
This could use a better name as well. It's unclear what `intermediateComponent` means without having to look at the implementation

Reply:
Since this `intermediateComponent` prop is only used in one place (DestinationForm), and it is just being placed after everything else in the ConnectorCard component, would it maybe make more sense to just remove the prop-drilling and place the `StartWithDestination` component after the `ConnectorCard` component in `DestinationForm`?

Reply:
Hmm... Yep, you are right - in this case, it's definitely better to move it to the bottom and remove the prop-drilling. 👍  

Since we had a talk with Nico and decided not to split the whole `ConnectorCard` into multiple `Card` components (leaving the old design) at least for now. 
So I will remove the `intermediateComponent`.

PS.
The reason why I called it `intermediateComponent` is because of the design: https://www.figma.com/file/8aM4gBoW95whyshI5kwVct/01_06_SOURCES?node-id=405%3A4015

Reply:
@dizel852 cool, once that change is made then I think this PR looks good to me from my end!